examples/ExecutionPools/AutoPart/test_plan.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+"""
+This example is to demonstrate auto-part feature in a process pool.
+"""
+
+import sys
+import typing
+from testplan import test_plan
+from testplan import Task
+from testplan.runners.pools.process import ProcessPool
+from testplan.common.utils.testing import calculate_pool_size_by_tasks
+from testplan.report.testing.styles import Style, StyleEnum
+
+OUTPUT_STYLE = Style(StyleEnum.ASSERTION_DETAIL, StyleEnum.ASSERTION_DETAIL)
+
+
+# Set the runtime limitation for auto-part tasks
+# Run testplan with --runtime-data runtime_data.json argument
+@test_plan(
+    name="AutoPartExample",
+    pdf_path="report.pdf",
+    stdout_style=OUTPUT_STYLE,
+    pdf_style=OUTPUT_STYLE,
+    auto_part_runtime_limit=100,
+)
+def main(plan):
+    """
+    Testplan decorated main function to add and execute MultiTests.
+
+    :return: Testplan result object.
+    :rtype:  ``testplan.base.TestplanResult``
+    """
+
+    # Discover tasks
+    tasks: typing.Collection[Task] = plan.discover(
+        path=".",
+        name_pattern=r".*task\.py$",
+    )
+
+    # Determine the number of pool size needed for running all the targets
```

Comment:
Calculate the right size of the pool based on the weight (runtime) of the tasks, so that runtime of all tasks meets the plan_runtime_target.